sound/soc/sof/control.c

```diff
@@ -134,25 +136,29 @@ int snd_sof_switch_put(struct snd_kcontrol *kcontrol,
 	struct snd_sof_dev *sdev = scontrol->sdev;
 	struct sof_ipc_ctrl_data *cdata = scontrol->control_data;
 	unsigned int i, channels = scontrol->num_channels;
+	int change = 0;
+	u32 value;
 
 	/* update each channel */
 	for (i = 0; i < channels; i++) {
-		cdata->chanv[i].value = ucontrol->value.integer.value[i];
+		value = ucontrol->value.integer.value[i];
+		change |= (value != cdata->chanv[i].value);
 		cdata->chanv[i].channel = i;
+		cdata->chanv[i].value = value;
 	}
 
 	if (scontrol->led_ctl.use_led)
```

Comment:
this led update could probably also not be called if nothing changes? but that's can be separate patch...

Reply:
Okay. I've also updated the commit to skip this when nothing changed.